### PR TITLE
 Allow to use the templating mechanism within a repetition

### PR DESCRIPTION
**Pull request by thibaultzanini**

### CODE REVIEW

File: core/serialization/serialization.js [hunk 1]
```diff
@@ -97,6 +97,20 @@ var Serialization = Montage.specialize( /** @lends Serialization.prototype # */
...
     },
 
+    removeObject: {
+        value: function (label) {
+            var serializationObject = this.getSerializationObject();
+
+            if (serializationObject && label in serializationObject) {
+                delete serializationObject[label];
+
+                this.initWithObject(serializationObject);
```
Comment (marchant) [inline core/serialization/serialization.js]: Is it needed to re-do a complete .initWithObject() It's unusual to do that. Performance impact? Side effects on current state?

File: core/template.js [hunk 2]
```diff
@@ -990,6 +990,26 @@ var Template = Montage.specialize( /** @lends Template# */ {
...
+                    if (parameterName === "*" && parameterElement.childElementCount > 0) { // has default
+                        if (argumentElement.childElementCount > 0 ||
+                            !!(argumentElement.firstChild &&
+                                argumentElement.firstChild.data &&
+                                argumentElement.firstChild.data.trim())
+                        ) { // has arguments
+                            var elementIds = this._getChildrenElementIds(parameterElement);
+                            
+                            for (var i = 0, length = elementIds.length; i < length; i++) {
+                                serialization.removeObject(elementIds[0]);
```
Comment (marchant) [inline core/template.js]: Each call to this.removeObject() will cause a this.initWithObject(serializationObject), sounds inefficient. Maybe we should have instead a removeObjets() that takes an array and do the initWithObject(serializationObject) only once at the end of the loop?

File: core/template.js [hunk 3]
```diff
@@ -990,6 +990,26 @@ var Template = Montage.specialize( /** @lends Template# */ {
...
+                            }
+                        } else {
+                            var range = parameterElement.ownerDocument.createRange();
```
Comment (marchant) [inline core/template.js]: I think Range can be reused, might be worth looking into that to minimize garbage creation

File: ui/component.js [hunk 4]
```diff
@@ -2327,26 +2327,30 @@ var Component = exports.Component = Target.specialize(/** @lends Component.proto
...
+                                this._element.firstChild.data &&
+                                this._element.firstChild.data.trim())
+                        ) {
                          //We're missing an argument, we're going to check if we have a default
-                             if (parameterElement && parameterElement.childElementCount > 0) {
-                                 range = this._element.ownerDocument.createRange();
-                                 range.selectNodeContents(parameterElement);
-                                 parameterElement.parentNode.replaceChild(range.extractContents(), parameterElement);
+                            if (parameterElement && parameterElement.childElementCount > 0) {
+                                range = this._element.ownerDocument.createRange();
```
Comment (marchant) [inline ui/component.js]: Similar comment about Range, if reuse works, we could create a pool thoughout montage

File: core/serialization/serialization.js (hunk 1 above)
Comment (thibaultzanini) [reply core/serialization/serialization.js]: Agreed maybe not the safest way of doing it .

File: core/template.js (hunk 2 above)
Comment (thibaultzanini) [reply core/template.js]: I provided a method called `removeObjectsWithLabels` for that purpose

File: ui/component.js (hunk 4 above)
Comment (thibaultzanini) [reply ui/component.js]: Not the goal of this PR, writing a new ticket

**Pull request merged**